app/app.go

```diff
@@ -219,6 +264,22 @@ func (a *app) GetDeviceIntegrations(
 }
 
 func (a *app) SetDeviceStatus(ctx context.Context, deviceID string, status model.Status) error {
+	if a.webhooksAsync {
+		a.webhooksQueue <- func() {
+			ctxWithTimeout, cancel := context.WithTimeout(ctx, a.webhooksTimeout)
```

Comment:
This won't work. The Context comming from api is cancelled once the request handler returns. A new Context should be created or at least the deadline should be stripped from the parent context.